wallet/net.go

```diff
@@ -0,0 +1,284 @@
+package wallet
+
+import (
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/textileio/textile-go/pb"
+	"golang.org/x/net/context"
+	"gx/ipfs/QmZoWKhxUmZ2seW4BzX6fJkNR8hh9PsGModr7q171yq2SS/go-libp2p-peer"
+	libp2pc "gx/ipfs/QmaPbCnUMBohSGo3KnxEa2bHqyJVVeEEcwtqJAYxerieBo/go-libp2p-crypto"
+	"gx/ipfs/QmcZfnkapfECQGcLZaf9B79NRg7cRa9EnZh4LSbkCzwNvY/go-cid"
+	"time"
+)
+
+//var OfflineMessageWaitGroup sync.WaitGroup
+
+func (w *Wallet) sendMessage(message *pb.Message, peerId string) error {
+	p, err := peer.IDB58Decode(peerId)
+	if err != nil {
+		return err
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	err = w.service.SendMessage(ctx, p, message)
+	if err != nil {
+		//if err := w.SendOfflineMessage(p, k, &message); err != nil {
+		return err
+		//}
+	}
+	return nil
+}
+
+// Supply of a public key is optional, if nil is instead provided n.EncryptMessage does a lookup
+//func (w *Wallet) SendOfflineMessage(p peer.ID, k *libp2pc.PubKey, m *pb.Message) error {
```

Comment:
(one more comment), this is the method we'll need for offline... see line 26 above, if there was an error reaching a peer, send offline instead. it would be nice to know if a peer is mobile, that way maybe you'd just send them offline always